src/uwtools/drivers/forecast.py

```diff
@@ -50,10 +50,15 @@ def batch_script(self) -> BatchScript:
         bs = self.scheduler.batch_script
         bs.append(pre_run)
         bs.append(self.run_cmd())
+        if self._dry_run:
+            log.info("Batch Script: ")
+            bs.dump(None)
```

Comment:
Thanks for this comment! It makes it much easier to understand the differences between ```stdout```, ```stdin```, and ```stderr```

Reply:
I've tried updating the respective dry run test using `assert logged(caplog, run_expected)` and I'm struggling to see why it's failing at the moment. I've pushed the most recent changes with the failing test.

Reply:
I'll fetch your branch and take a look -- more soon.

Reply:
I think what's happening is that `logged()` determines whether its `msg` argument appears in _some_ line of logged output (in some log "record" in logging parlance). But `run_expected` is a big string with _all_ the lines of logged output, joined with newline characters.

This works for me:
```
% git diff -U0
diff --git a/src/uwtools/tests/drivers/test_forecast.py b/src/uwtools/tests/drivers/test_forecast.py
index 125b8e0..1b9b7bb 100644
--- a/src/uwtools/tests/drivers/test_forecast.py
+++ b/src/uwtools/tests/drivers/test_forecast.py
@@ -347 +347 @@ def test_FV3Forecast_run_dry_run(caplog, fv3_mpi_assets, fv3_run_assets, with_ba
-        run_expected = "\n".join(batch_components)
+        run_expected = batch_components
@@ -350 +350 @@ def test_FV3Forecast_run_dry_run(caplog, fv3_mpi_assets, fv3_run_assets, with_ba
-        run_expected = " ".join(fv3_mpi_assets)
+        run_expected = [" ".join(fv3_mpi_assets)]
@@ -356,2 +356,2 @@ def test_FV3Forecast_run_dry_run(caplog, fv3_mpi_assets, fv3_run_assets, with_ba
-    assert logged(caplog, run_expected)
-    # assert run_expected in capsys.readouterr().out
+    for line in run_expected:
+        assert logged(caplog, line)
```

Reply:
Maybe `expected_lines` would be a good name instead of `run_expected`, to emphasize the fact that it should contain plural lines, suggesting a `list` object.